### PR TITLE
   mon/PGMap: Fix the incorrect deletion of map element

### PR DESCRIPTION
**Pull request by taodd**
Fixes: https://tracker.ceph.com/issues/61912
    Signed-off-by: dongdong tao <dongdong.tao@canonical.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/mon/PGMap.cc [hunk 1]
```diff
@@ -1212,10 +1212,12 @@ void PGMap::apply_incremental(CephContext *cct, const Incremental& inc)
...
       osd_stat.erase(t);
     }
-    for (auto i = pool_statfs.begin();  i != pool_statfs.end(); ++i) {
+    for (auto i = pool_statfs.begin();  i != pool_statfs.end(); ) {
       if (i->first.second == *p) {
 	pg_pool_sum[i->first.first].sub(i->second);
-	pool_statfs.erase(i);
+	i = pool_statfs.erase(i);
+      } else {
+	  ++i;
```
Comment (Matan-B) [inline src/mon/PGMap.cc:1220]: indentation

**Review by Matan-B — APPROVED** [review]

File: src/mon/PGMap.cc (hunk 1 above)
Comment (taodd) [reply src/mon/PGMap.cc]: thanks, fixed

**Comment by Matan-B** [comment]
Can you please rename the commit title to mon/PGMap?

**Comment by taodd** [comment]
> Can you please rename the commit title to mon/PGMap?

done

**Comment by yukariatlas** [comment]
It's a pure C++ iteraotr issue, LGTM!

**Review by pponnuvel — APPROVED** [review]
LGTM. 

I tried few things to see if I can get g++ or clang++ to warn on these things. Alas, no luck even with `-fsanitize=address,undefined`.

**Comment by Matan-B** [comment]
@taodd, Thank you for the PR.
I noticed that we have an older pending PR which addresses the issue as well, I will close this in favor of #49480.

**Pull request closed without merging**